projects/ephemery-genesis-reset-implementation.md

```diff
@@ -0,0 +1,81 @@
+
+Currently, this is implemented as external tooling for clients, wherefore, resetting the testnet requires stopping the client and then removing the chaindata manually. Ideally, clients should be able to automatically reset themselves once the ephemery network has progressed to a new genesis state. 
+
+Holly, a fellow also working on Ephemery, is currently implementing the [genesis function](https://github.com/eth-protocol-fellows/cohort-four/blob/master/projects/native-ephemery-genesis.md) on Lodestar. This will offer partial client support for running the current instance of the Ephemery network. An automatic reset implementation that complements this, will ensure full client support!
+
+This is the next phase of ephemery and it's what I will be working on for my project. The immediate benefit of this is that: it will provide a more efficient testing environment for developers & validators making it easier to validate, and experiment with [Ephemery](https://ephemery.dev/).
+
+## Project description
+The scope of this project will be focussed on: 
+
+1. Automatic Reset Implementation: Develop and integrate the `Automatic Reset` feature into the Lodestar Ethereum client. The first step in achieving this is to understand Lodestar's architecture; vis-a-vis network handling & use of consensus specs. Then use that to implement the following process in code: 
+
+    * Database Cleanup
+    * Update client configurations
+    * Restart client (Lodestar)
+    * Sync with new genesis (Ephemery)
+    * Update network parameters like network ids, peers, etc
+    * Update monitoring tools like grafana
+    * Restart staking & validation
+
+2. Genesis State Management: Developing programming logic that defines a clear process for managing the genesis state during resets as outlined in the [specifications](https://github.com/ephemery-testnet/ephemery-resources/blob/master/specs.md). This will include how clients handle:
+    * the deletion or archiving of chaindata;
+    * and generating a new genesis state with updated parameters.
+
+- I will be collaborating with Holly directly on this part of the project.
+
+3. Validation and Testing: Thoroughly validate the implementation to ensure its compatibility with the Ephemery testnet's specifications. Conduct extensive testing to identify and resolve any potential issues.
+
+4. Documentation and User Support: Create comprehensive documentation for validators and developers, explaining how to interact with the Ephemery testnet and leverage the Automatic Reset feature.
+
+
+
+**Stretch 01:** Implement safety checks to prevent accidental resets, by adding an optional flag with a tag like `--require-manual-reset`
+
+**Stretch 02:** Add another optional flag `--archive chaindata` for genesis management.
```

Comment:
So using this flag, you would keep ancients from all previous iterations? Even if you would do that, the client doesn't really have a way of accessing them so another mechanism would need to be added. Seems like a lot of complexity but feel free to elaborate.